lib/split/trial.rb

```diff
@@ -63,12 +77,12 @@ def choose!(context = nil)
       elsif @experiment.has_winner?
         self.alternative = @experiment.winner
       else
-        cleanup_old_versions
+        #cleanup_old_versions
 
         if exclude_user?
           self.alternative = @experiment.control
         else
-          value = @user[@experiment.key]
+          value = @user.active_experiments[@experiment.name]
```

Comment:
What's the difference between these two?

Reply:
active_experiments returns experiments for which there is no winner 
```
    def active_experiments
      experiment_pairs = {}
      keys_without_finished(user.keys).each do |key|
        Metric.possible_experiments(Split::Experiment.key_without_version_for_experiment_key(key)).each do |experiment|
          if !experiment.has_winner?
            experiment_pairs[Split::Experiment.key_without_version_for_experiment_key(key)] = user[key]
          end
        end
      end
      experiment_pairs
    end
```